redash/query_runner/pinot.py

```diff
@@ -0,0 +1,135 @@
+from redash.query_runner import TYPE_DATETIME, TYPE_FLOAT, TYPE_STRING, TYPE_INTEGER, TYPE_BOOLEAN
+from redash.utils import json_dumps
+
+import requests
+from requests.auth import HTTPBasicAuth
+import logging
+
+logger = logging.getLogger(__name__)
+
+PINOT_TYPES_MAPPING = {
+    "BOOLEAN": TYPE_BOOLEAN,
+    "INT": TYPE_INTEGER,
+    "LONG": TYPE_INTEGER,
+    "FLOAT": TYPE_FLOAT,
+    "DOUBLE": TYPE_FLOAT,
+    "STRING": TYPE_STRING,
+    "BYTES": TYPE_STRING,
+    "JSON": TYPE_STRING,
+    "TIMESTAMP": TYPE_DATETIME,
+}
+
+
+class Pinot(BaseQueryRunner):
+    noop_query = "SELECT 1"
+    controller_uri = "http://localhost:9000"
+    username = None
+    password = None
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "brokerHost": {"type": "string", "default": "localhost"},
+                "brokerPort": {"type": "number", "default": 8099},
+                "brokerScheme": {"type": "string", "default": "http"},
+                "controllerURI": {"type": "string", "default": "http://localhost:9000"},
+                "username": {"type": "string"},
+                "password": {"type": "string"},
+            },
+            "order": ["brokerScheme", "brokerHost", "brokerPort", "controllerURI", "username", "password"],
+            "required": ["brokerHost", "controllerURI"],
+            "secret": ["password"],
+        }
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    def __init__(self, configuration):
+        super(Pinot, self).__init__(configuration)
+        self.controller_uri = self.configuration.get("controllerURI")
+        self.username=(self.configuration.get("username") or None)
+        self.password=(self.configuration.get("password") or None)
+
+    def run_query(self, query, user):
+        logger.debug("Running query %s with username: %s, password: %s", query, self.username, self.password)
+        connection = pinotdb.connect(
+            host=self.configuration["brokerHost"],
+            port=self.configuration["brokerPort"],
+            path="/query/sql",
+            scheme=(self.configuration.get("brokerScheme") or "http"),
+            verify_ssl=False,
+            username=self.username,
+            password=self.password,
+        )
+
+        cursor = connection.cursor()
+
+        try:
+            cursor.execute(query)
+            logger.debug("cursor.schema = %s",cursor.schema)
+            columns = self.fetch_columns(
+                [(i["name"], PINOT_TYPES_MAPPING.get(i["type"], None)) for i in cursor.schema]
+            )
+            rows = [
+                dict(zip((column["name"] for column in columns), row)) for row in cursor
+            ]
+
+            data = {"columns": columns, "rows": rows}
+            error = None
+            json_data = json_dumps(data)
+            logger.debug("Pinot execute query [%s], and resp: %s", query, json_data)
```

Comment:
Does it make sense to log out the entire query result? Especially for large results this will balloon the size of the log files unnecessarily. It's sufficient to log out the query text alone.
```suggestion
            logger.debug("Pinot execute query [%s]", query)
```